brvs-core/src/integration-test/java/iroha/validation/behavior/IrohaIntegrationTest.java

```diff
@@ -817,4 +826,86 @@ void exactDistributionPortions() throws InterruptedException {
     assertNotNull(finishedNow);
     assertTrue(finishedNow.getFinished());
   }
+
+  /**
+   * @given {@link ValidationService} instance with {@link SoraDistributionPluggableLogic} attached
+   * and relevant JSON with Sora distribution proportions provided
+   * @when {@link Transaction} with {@link iroha.protocol.Commands.Command TransferAsset} command
+   * going from a project owner appears
+   * @then {@link ValidationService} performs nothing since it has not enough balance
+   */
+  @Test
+  void insufficientDistributionBalance() throws InterruptedException {
+    // projectOwner is a json setter
+    final Map<String, BigDecimal> proportionsMap = new HashMap<>();
+    proportionsMap.put(projectParticipantOneId, new BigDecimal("0.005"));
+    proportionsMap.put(projectParticipantTwoId, new BigDecimal("0.003"));
+    proportionsMap.put(projectParticipantThreeId, new BigDecimal("0.002"));
+    final BigDecimal totalSupply = new BigDecimal("100000");
+    final SoraDistributionProportions proportions = new SoraDistributionProportions(
+        proportionsMap,
+        totalSupply
+    );
+    final QueryAPI queryAPI = new QueryAPI(irohaAPI, validatorId, validatorKeypair);
+
+    BigDecimal oneBalance = new BigDecimal(queryAPI.getAccountAssets(projectParticipantOneId)
```

Comment:
The same code to retrieve balance is used 6 times, let move it to function or separate method